### PR TITLE
fix(templates): keep element name after remove

### LINKED ISSUE #570: Removing an element template clears its name
__Describe the Bug__

If I remove a linked element template the activities name is cleared:

capture IpqPNi_optimized

As a user I do not expect the name to be part of the element template, thus keeping it.

__Steps to Reproduce__

1. Load [element-templates.bpmn](https://github.com/bpmn-io/bpmn-js-properties-panel/files/8014624/element-templates.bpmn.txt)
2. Remove template from any element
3. See that name is removed with it

__Expected Behavior__

Name like elements that I can continue to customize as I have a template applied (group categroy value?, name, text annotation body?) are not reset as the template is being removed.

__Environment__

- Host (Browser/Node version), if applicable: Any
- OS: Any
- Library version: bpmn-js-properties-panel@1.0.0-alpha.2, `#master` (both cloud and platform providers affected)

### PR DESCRIPTION
**Pull request by marstamm**
closes #570

Recording 2022-04-11 at 11 28 14

### CODE REVIEW

File: src/provider/element-templates/util/templateUtil.js [hunk 1]
```diff
@@ -14,18 +15,24 @@ export function unlinkTemplate(element, injector) {
...
+        eventDefinitionType = getEventDefinitionType(businessObject),
+        label = getLabel(element);
 
   const newElement = replace.replaceElement(element, {
     type: type,
     eventDefinitionType: eventDefinitionType
   });
 
+  if (label) {
+    modeling.updateLabel(newElement, label);
```
Comment (pinussilvestrus) [inline src/provider/element-templates/util/templateUtil.js:33]: As we are firing a command here, the next undo after removing the template will remove the label

Kapture 2022-04-11 at 11 40 51

Only after the second undo the remove is undone. I'd expect that `undo` after the removal brings me back the template immediately.

**Review by pinussilvestrus — CHANGES_REQUESTED** [review]
We should make sure that `undo` works properly.

**Review by nikku — COMMENTED** [review]
Did we properly root cause this one?

With stock bpmn-js the name is kept on replace, cf.: 

capture mF96yC_optimized

How do we accomplish that (and why does it not work in connectors land)? Please root cause and we find a robust solution.

File: src/provider/element-templates/util/templateUtil.js (hunk 1 above)
Comment (nikku) [reply src/provider/element-templates/util/templateUtil.js:33]: See also https://github.com/bpmn-io/bpmn-js-properties-panel/pull/662#pullrequestreview-937755048.

File: src/provider/element-templates/util/templateUtil.js [hunk 2]
```diff
@@ -16,23 +16,23 @@ export function unlinkTemplate(element, injector) {
...
-        modeling = injector.get('modeling');
+        bpmnFactory = injector.get('bpmnFactory');
 
   const businessObject = getBusinessObject(element);
 
   const type = businessObject.$type,
         eventDefinitionType = getEventDefinitionType(businessObject),
         label = getLabel(element);
 
+  const newBusinessObject = bpmnFactory.create(businessObject.$type);
```
Comment (nikku) [inline src/provider/element-templates/util/templateUtil.js:27]: We need to be aware of different names here:

* Text annotaiton, group have different _names_

Comment (nikku) [reply src/provider/element-templates/util/templateUtil.js:27]: Other than that I like this direction.

Comment (marstamm) [reply src/provider/element-templates/util/templateUtil.js]: Groups are not that straight forward, but should be all good now:
https://github.com/bpmn-io/bpmn-js-properties-panel/blob/905eaf8e2b82fcd9b20925ac20b6abe30184de8d/src/provider/element-templates/util/templateUtil.js#L112-L118

File: src/provider/element-templates/util/templateUtil.js [hunk 3]
```diff
@@ -89,3 +94,30 @@ function leftPad(string, length, character) {
...
+
+  const bo = getBusinessObject(element),
+        newBo = bpmnFactory.create(bo.$type),
+        label = getLabel(element);
+
+  if (!label) {
+    return newBo;
+  }
+
+  if (is(element, 'bpmn:TextAnnotation')) {
```
Comment (nikku) [inline src/provider/element-templates/util/templateUtil.js:110]: You could use [`LabelUtil#setLabel`](https://github.com/bpmn-io/bpmn-js/blob/develop/lib/features/label-editing/LabelUtil.js#L52) here.

Comment (nikku) [reply src/provider/element-templates/util/templateUtil.js:110]: And wrap it in a `try/catch` for good measure to handle categoryRef not existing, of course.

Comment (marstamm) [reply src/provider/element-templates/util/templateUtil.js:110]: > And wrap it in a try/catch for good measure to handle categoryRef not existing, of course

Ah, slow Github updates :smile:

Comment (marstamm) [reply src/provider/element-templates/util/templateUtil.js]: Done in [4b961e0](https://github.com/bpmn-io/bpmn-js-properties-panel/pull/662/commits/4b961e092266ca77bb9784b3faf88818c67b44a0)

**Review by nikku — APPROVED** [review]
:clap:

**Pull request merged**